src/main/java/com/aws/greengrass/deployment/IotJobsClientWrapper.java

```diff
@@ -172,85 +202,116 @@ public CompletableFuture<Integer> PublishDescribeJobExecution(DescribeJobExecuti
+                                            exceptionHandler.accept(e);
+                                        }
+                                    }
+                                });
         return this.connection.subscribe(topic, qos, messageHandler);
     }
 
     @Override
     public CompletableFuture<Integer> SubscribeToJobExecutionsChangedEvents(
-            JobExecutionsChangedSubscriptionRequest request, QualityOfService qos,
-            Consumer<JobExecutionsChangedEvent> handler, Consumer<Exception> exceptionHandler) {
+            JobExecutionsChangedSubscriptionRequest request,
+            QualityOfService qos,
+            Consumer<JobExecutionsChangedEvent> handler,
+            Consumer<Exception> exceptionHandler) {
 
         if (request.thingName == null) {
             CompletableFuture<Integer> result = new CompletableFuture();
-            result.completeExceptionally(new MqttException(
-                    "JobExecutionsChangedSubscriptionRequest must have a non-null thingName"));
+            result.completeExceptionally(
+                    new MqttException(
+                            "JobExecutionsChangedSubscriptionRequest must have a non-null thingName"));
             return result;
         }
 
         String topic = String.format(JOB_EXECUTIONS_CHANGED_TOPIC, request.thingName);
         Consumer<MqttMessage> messageHandler =
-                jobExecutionCbs.computeIfAbsent(new Pair<>(handler, exceptionHandler), (k) -> (message) -> {
-                    try {
-                        String payload = new String(message.getPayload(), StandardCharsets.UTF_8);
-                        JobExecutionsChangedEvent response =
-                                this.gson.fromJson(payload, JobExecutionsChangedEvent.class);
-                        handler.accept(response);
-                    } catch (Exception e) {
-                        if (exceptionHandler != null) {
-                            exceptionHandler.accept(e);
-                        }
-                    }
-                });
+                jobExecutionCbs.computeIfAbsent(
+                        new Pair<>(handler, exceptionHandler),
+                        (k) ->
+                                (message) -> {
+                                    try {
+                                        String payload =
+                                                new String(
+                                                        message.getPayload(),
+                                                        StandardCharsets.UTF_8);
+                                        JobExecutionsChangedEvent response =
+                                                this.gson.fromJson(
+                                                        payload, JobExecutionsChangedEvent.class);
+                                        handler.accept(response);
+                                    } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that your code handles a broad swath of exceptions in the catch block, potentially trapping dissimilar issues or problems that should not be dealt with at this point in the program.

[Learn more](https://cwe.mitre.org/data/definitions/396.html)